lib/cartopy/mpl/patch.py

```diff
@@ -185,13 +185,10 @@ def path_to_geos(path, force_ccw=False):
                                      axis=1)
         if all(verts_same_as_first):
             geom = sgeom.Point(path_verts[0, :])
-        elif (path_verts.shape[0] > 2 and
+        elif (path_verts.shape[0] > 3 and
                 (path_codes[-1] == Path.CLOSEPOLY or
                  verts_same_as_first[-1])):
-            if path_codes[-1] == Path.CLOSEPOLY:
```

Comment:
For `CLOSEPOLY`, the last location should be ignored; see [Matplotlib's Path tutorial](https://matplotlib.org/users/path_tutorial.html). Thus, I don't think this condition should be removed, unless you can guarantee that the last point is at the first point (which seems unlikely given the `or verts_same_as_first[-1]` bit.)